extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -165,41 +122,58 @@ func CreateRKE2K3SSnapshot(client *rancher.Client, clusterName string) error {
 	}
 
 	logrus.Infof("Creating snapshot...")
-	_, err = client.Steve.SteveType(clusters.ProvisioningSteveResourceType).Update(clusterSteveObject, clusterObject)
+	updatedCluster, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).Update(clusterSteveObject, clusterObject)
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	updateTimestamp := updatedCluster.ManagedFields[len(updatedCluster.ManagedFields)-1].Time.Time
+
+	updateTimestamp = time.Now()
+	err = clusters.WaitOnClusterAfterSnapshot(client, updatedCluster.ID)
+	if err != nil {
+		return nil, err
 	}
 
+	var snapshots []rancherv1.SteveAPIObject
+
 	err = kwait.PollUntilContextTimeout(context.TODO(), 5*time.Second, defaults.FiveMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
-		snapshotSteveObjList, err := client.Steve.SteveType("rke.cattle.io.etcdsnapshot").List(nil)
+		query, err := url.ParseQuery(fmt.Sprintf("labelSelector=%s=%s", SnapshotClusterNameLabel, clusterName))
 		if err != nil {
 			return false, nil
 		}
 
-		_, clusterSteveObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+		snapshotSteveObjList, err := client.Steve.SteveType(SnapshotSteveResourceType).List(query)
 		if err != nil {
 			return false, nil
 		}
+		if len(snapshotSteveObjList.Data) == 0 {
+			return false, nil
+		}
 
+		snapshots = []rancherv1.SteveAPIObject{}
 		for _, snapshot := range snapshotSteveObjList.Data {
-			snapshotObj, err := client.Steve.SteveType("rke.cattle.io.etcdsnapshot").ByID(snapshot.ID)
+			_, err = client.Steve.SteveType(SnapshotSteveResourceType).ByID(snapshot.ID)
 			if err != nil {
 				return false, nil
 			}
 
-			if snapshotObj.ObjectMeta.State.Name == active && clusterSteveObject.ObjectMeta.State.Name == active {
-				logrus.Infof("All snapshots in the cluster are in an active state!")
-				return true, nil
+			// snapshot time doesn't include nanoseconds, but time.Now() does. Rounding up by 1 Second.
+			if snapshot.CreationTimestamp.Time.Add(time.Duration(time.Second)).Compare(updateTimestamp) > -1 {
+				snapshots = append(snapshots, snapshot)
 			}
 		}
 
-		return false, nil
+		if len(snapshots) == 0 {
+			return false, nil
+		}
+
+		return true, nil
 	})
-	if err != nil {
-		return err
-	}
 
-	return nil
+	// not registering cleanup func; users do not delete snapshots through rancher
```

Comment:
This can probably be removed.

Reply:
I actually prefer to keep this in, since it isn't following the same pattern as other functions. is it okay with you to keep the comment?